jf_agent/data_manifests/git/generator.py

```diff
@@ -22,12 +21,41 @@ class UnsupportedGitProvider(Exception):
 
 
 def create_manifests(
-    company_slug: str, creds: dict, config: dict, verbose: bool = False
+    company_slug: str,
+    creds: dict,
+    config: dict,
+    endpoint_git_instances_info: dict,
+    verbose: bool = False,
 ) -> list[GitDataManifest]:
     manifests: list[GitDataManifest] = []
     git_configs: list[GitConfig] = config.git_configs
+    is_multi_git_config = len(git_configs) > 1
+
+    # Iterate over each git config within the config yaml
     for git_config in git_configs:
+        if is_multi_git_config:
+            instance_slug = git_config.git_instance_slug
+            instance_info = endpoint_git_instances_info.get(instance_slug)
+            instance_creds = creds.git_instance_to_creds.get(instance_slug)
+        else:
+            # support legacy single-git support, which assumes only one available git instance
+            instance_info = list(endpoint_git_instances_info.values())[0]
+            instance_creds = list(creds.git_instance_to_creds.values())[0]
+
         try:
+            instance_slug: str = instance_info['slug']
```

Comment:
Q: Why do we retrieve the instance slug here? In the case of legacy single-git, this makes sense, but for multi-get we'd be re-retrieving the slug and replacing the one we get from our git_config.

I assume this won't actually cause any issue (since the one we get back from the endpoint_info will be the same), just wondering about the thinking behind this.

Reply:
The main reason is that I was mirroring the existing agent logic as much as possible here. The secondary reason is that I think it'd better to have the 'truth' values come from the jellyfish app, rather than the the config.yaml. There could be one day where we need to overwrite the slugs in the yaml for some reason (very unlikely, but you never know), so I think it makes sense to keep it this way even if it is a little redundant

That being said, I should really dedup this logic between the existing git ingestion logic... I'll do that now